### PR TITLE
messages: remove compat cruft

### PR DESCRIPTION
**Pull request by liewegas**
Remove compat cruft from server to server messages for pre-jewel daemons.

### CODE REVIEW

File: src/osd/osd_types.cc [hunk 1]
```diff
@@ -1928,111 +1928,46 @@ void object_stat_sum_t::encode(bufferlist& bl) const
...
-  DECODE_START_LEGACY_COMPAT_LEN(14, 3, 3, bl);
+  DECODE_START(15, bl);
 #if defined(CEPH_LITTLE_ENDIAN)
-  if (struct_v >= 15) {
-    bl.copy(sizeof(object_stat_sum_t), (char*)(&num_bytes));
-    decode_finish = true;
-  }
+  bl.copy(sizeof(object_stat_sum_t), (char*)(&num_bytes));
+  decode_finish = true;
 #endif
```
Comment (branch-predictor) [inline src/osd/osd_types.cc]: How about removing the decode_finish variable and use "#else" instead?

File: src/messages/MOSDPGQuery.h [hunk 2]
```diff
@@ -63,34 +63,47 @@ class MOSDPGQuery : public Message {
...
-      pg_list.insert(
-	make_pair(
-	  spg_t(_pg_list[i].first, _shard_list[i]), _pg_list[i].second));
+    if (header.version < 4) {
+      // for kraken/jewel only
+      ::decode(epoch, p);
+      vector<pair<pg_t, pg_query_t> > _pg_list;
+      ::decode(_pg_list, p);
+      vector<shard_id_t> _shard_list(_pg_list.size(), shard_id_t::NO_SHARD);
+      _shard_list.clear();
```
Comment (tchaikov) [inline src/messages/MOSDPGQuery.h]: why not just `_shard_list.reserve()`?

**Comment by liewegas** [comment]
TBH i didn't even read this code, just put it inside the if block so we could remove it post-luminous.

File: src/osd/osd_types.cc (hunk 1 above)
Comment (liewegas) [inline src/osd/osd_types.cc]: i'm adding the if() back in so that it's clear how to add new fields next time it happens.

**Comment by liewegas** [comment]
rebased

**Comment by liewegas** [comment]
2017-04-22T01:02:38.287 INFO:tasks.ceph.osd.4.smithi174.stderr:2017-04-22 01:02:30.318304 7f116f018700 -1 failed to decode message of type 80 v5: buffer::malformed_input: void object_stat_sum_t::decode(ceph::buffer::list::iterator&) unknown encoding version > 14


/a/sage-2017-04-21_23:56:14-rados-wip-sage-testing---basic-smithi/1054134

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-04-25_02:25:56-rados-wip-sage-testing---basic-smithi/

test looks good, needs review

**Comment by tchaikov** [comment]
will take a look tmr.

**Comment by liewegas** [comment]
@tchaikov ping

**Comment by tchaikov** [comment]
ahh, i promise, i will review **tomorrow**. running out of time today.

File: src/messages/MOSDPGQuery.h [hunk 3]
```diff
@@ -62,44 +63,47 @@ class MOSDPGQuery : public Message {
...
+    ::encode(pg_list, payload, features);
   }
   void decode_payload() override {
     bufferlist::iterator p = payload.begin();
-    ::decode(epoch, p);
-    vector<pair<pg_t, pg_query_t> > _pg_list;
-    ::decode(_pg_list, p);
-    vector<shard_id_t> _shard_list(_pg_list.size(), shard_id_t::NO_SHARD);
-    if (header.version >= 3) {
+    if (header.version < 4) {
```
Comment (tchaikov) [inline src/messages/MOSDPGQuery.h]: nit, could be more specific, like
```c
if (header.version == 3) {
```

**Comment by tchaikov** [comment]
modulo the nit, lgtm.

**Comment by liewegas** [comment]
thanks!

**Pull request merged**